libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/PushNotificationManager.swift

```diff
@@ -84,3 +89,100 @@ extension PushNotificationManager {
         }
     }
 }
+
+extension PushNotificationManager {
+    private func fetchNotificationTypesFromAPI(with client: RestClient) async throws -> [NotificationType] {
+        let request = RestRequest(method: .GET, path: "connect/notifications/types", queryParams: nil)
+        let response = try await client.send(request: request)
+        let result = try response.asDecodable(type: NotificationTypesResponse.self)
+        return result.notificationTypes
+    }
+    
+    private func storeNotification(types: [NotificationType], with userDefaults: UserDefaults) {
+        do {
+            let encodedData = try JSONEncoder().encode(types)
+            userDefaults.set(encodedData, forKey: UserDefaultsKeys.cachedNotificationTypes)
```

Comment:
@wmathurin  I'm storing the Notification types in User Defaults.  Is there another place we want to store these?  OR is this acceptable?

Reply:
We support multiple users logged in at the same time, so we have dedicated directories for every user, and persisted for a given user must go in the corresponding directory.
Also we encrypt everything at rest.

You should take a look at [SFDefaultUserAccountPersister.m](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/dev/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/UserAccount/SFDefaultUserAccountPersister.m).

Maybe the notification stuff should be part of the [SFUserAccount](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/dev/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/UserAccount/SFUserAccount.h) alongside the credentials, identity data etc ?

Reply:
Made the change the notification stuff is stored on the UserAccount instead now.